test/e2e/common/kubelet_etc_hosts.go

```diff
@@ -137,6 +139,27 @@ func (config *KubeletManagedHostConfig) getEtcHostsContent(podName, containerNam
 }
 
 func (config *KubeletManagedHostConfig) createPodSpec(podName string) *v1.Pod {
+
+	etcHostContents := `127.0.0.1	localhost
+::1	localhost ip6-localhost ip6-loopback
+fe00::0	ip6-localnet
+ff00::0	ip6-mcastprefix
+ff02::1	ip6-allnodes
+ff02::2	ip6-allrouters`
+
+	// Write the data to a temp file.
+	tmpfile, err := ioutil.TempFile("", "etc-hosts")
```

Comment:
how does writing a random temp file affect the test? this doesn't appear to be used...

Reply:
would we only want to do this in some environments? the referenced issue was specific to the local-up-cluster conformance job, right?

Reply:
sorry pushed up an update. 

The /etc/hosts file in the environment where pods are launched from is just mounted to the pod. And then there is a check if the file has the string that shows if the file was a managed one or not (using cat on the launched container). So in the CI environment, the /etc/hosts already has the string. so we now create a fresh hosts file and use that.